source/AI.cpp

```diff
@@ -3098,6 +3098,14 @@ void AI::AutoFire(const Ship &ship, FireCommand &command, bool secondary) const
 			if(!secondary || fuel < (isStaying ? 0. : ship.JumpNavigation().JumpFuel()))
 				continue;
 		}
+
+		// Weapons that shoot ships are always reaching the target:
+		if(!weapon->ShipToShoot().empty() && !enemies.empty())
```

Comment:
I'd say use a new method for this.
That way how the weapon handles it is done internaly and it could very well be looking trough submunitions looking for a ship.

Reply:
how would you call it?

Reply:
Actually nvm just remove this and let CC define a custom "range override" for these instead of always shooting?

Reply:
ok